Clean up unused @ts-expect-error directives in collections components

## Issue Summary

Multiple files contain unused `@ts-expect-error` directives that should be removed to improve code quality.

## TypeScript Errors

```
app/collections/[slug]/opengraph-image.tsx(44,7): error TS2578: Unused '@ts-expect-error' directive.

app/collections/[slug]/trends/content.tsx(25,1): error TS2578: Unused '@ts-expect-error' directive.
```

## Impact

- **Code quality**: Unused directives indicate outdated workarounds or resolved issues
- **Maintainability**: Future developers may be confused by unnecessary directives
- **Type safety**: May mask real issues if the underlying problem was never fixed

## Files Affected

1. `apps/web/app/collections/[slug]/opengraph-image.tsx` (line 44)
2. `apps/web/app/collections/[slug]/trends/content.tsx` (line 25)

## Suggested Fix

1. Review each `@ts-expect-error` directive to understand what issue it was meant to address
2. If the underlying issue is resolved, remove the directive
3. If the issue persists, fix the root cause instead of suppressing the error

### Example for opengraph-image.tsx

```typescript
// Before (line 44)
// @ts-expect-error - some old workaround

// After
// (remove the directive if no longer needed)
```

### Example for trends/content.tsx

```typescript
// Before (line 25)
// @ts-expect-error - some old workaround

// After
// (remove the directive if no longer needed)
```

## Priority

**Low** - Code quality improvement, does not affect runtime

## References

- Similar to #2344 (unused `@ts-expect-error` in sitemap.ts)

--- a/apps/web/app/sitemap.ts
+++ b/apps/web/app/sitemap.ts
@@ -28,7 +28,6 @@ export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
       url: `${SITE_URL}${page.path}`,
       changeFrequency: 'weekly',
       priority: page.priority,
-      // @ts-expect-error — Next.js MetadataRoute.Sitemap doesn't type images yet
       ...(page.images ? { images: page.images } : {}),
     });
   }
@@ -42,7 +41,6 @@ export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
         url: `${SITE_URL}/collections/${slug}`,
         changeFrequency: 'weekly',
         priority: 0.6,
-        // @ts-expect-error — Next.js MetadataRoute.Sitemap doesn't type images yet
         images: [`${SITE_URL}/collections/${slug}/opengraph-image`],
       });
       entries.push({

--- a/apps/web/charts/analyze/org/company/visualization.tsx
+++ b/apps/web/charts/analyze/org/company/visualization.tsx
@@ -1,6 +1,5 @@
 import type { WidgetVisualizerContext } from '@/lib/charts-types';
 
-// @ts-expect-error d3-cloud has no type declarations
 import cloud from 'd3-cloud';
 
 type Params = {

--- a/apps/web/charts/analyze/repo/company/visualization.tsx
+++ b/apps/web/charts/analyze/repo/company/visualization.tsx
@@ -1,6 +1,5 @@
 import type { WidgetVisualizerContext } from '@/lib/charts-types';
 // import { compare } from '@/lib/charts-utils/visualizer/analyze';
-// @ts-expect-error d3-cloud has no type declarations
 import cloud from 'd3-cloud';
 // import xss from 'xss';
 

